rp_asos/models.py

```diff
@@ -67,10 +67,9 @@ def get_wa_group_info(self):
             group_info["id"] = self.whatsapp_group_id
         return group_info
 
-    def send_group_invites(self, group_info, msisdns):
+    def send_group_invites(self, group_info, wa_ids):
```

Comment:
From my understanding, this is the non-idempotent part / the part that we can't just repeat if the task fails in the middle (or at least, line 81 `utils.send_whatsapp_template_message`). 

If that's the case, it might be better to separate just the `send_whatsapp_template_message` into a separate task, and instead of executing it here, queueing it to be executed. 

That way we can retry this task as many times as we want and be safe.

Reply:
The problem with that is I'll hit the limit every single time on the main task then. 

I think it's around 8 request per hospital without the send call, multiplied by 30 hospitals in SA it's well over the 60 per minute.

Reply:
Ah I see. I was thinking of keeping this split, but then also splitting it further, otherwise if we rerun this we might be sending duplicate messages, so it's not really safe to retry

Reply:
I see, I think I can rework that a bit.